client/my-sites/themes/theme-activation-confirmation-modal.jsx

```diff
@@ -82,9 +85,9 @@ class AutoLoadingHomepageModal extends Component {
 	closeModalHandler = ( action = 'dismiss' ) => () => {
 		const { installingThemeId, siteId, source } = this.props;
 		if ( 'activeTheme' === action ) {
-			this.props.acceptAutoLoadingHomepageWarning( installingThemeId );
+			this.props.acceptActivateModalWarning( installingThemeId );
 			const keepCurrentHomepage = this.state.homepageAction === 'keep_current_homepage';
-			recordTracksEvent( 'calypso_theme_autoloading_homepage_modal_activate_click', {
+			recordTracksEvent( 'calypso_theme_activation_confirmation_modal_activate_click', {
```

Comment:
Changing this event name will disconnect it from data we've collected since #52926 was deployed (just last week, so not much).

But this does make me wonder if we should have a more generic event name and use props to track the type of modal we're viewing (`retired` theme warning, `auto-loading homepage`, etc.) /cc @kwight @ianstewart any preferences here?

Reply:
I don't think we need to go there – we can always use unique names (even if they get a little longer).

Reply:
OK, so it sounds like we should update this and record a separate Tracks event for the retired theme modal rather than changing the autoloading homepage event.

Reply:
Sure thing, but assuming that this design is okay, which one should be fired when the modal in https://github.com/Automattic/wp-calypso/pull/40793#issuecomment-848902352 is displayed?

Reply:
I think we can leave all the `autoloading_homepage` Tracks events as-is, and add a new event for the retired theme warning (a `*_view` event when the retired theme warning message is shown on the autoloading homepage modal). Then when looking at Tracks flows we can segment by folks who've seen both the autoloading homepage modal AND the retired theme warning at the same time.

I don't think it's necessary to add separate `click` and `dismiss` events to the buttons when the warning is displayed, though.

Let me know if that makes sense. :D

Reply:
I think that I've implemented those changes now - let me know if I've misunderstood! :)